CMakeLists.txt

```diff
@@ -67,68 +330,184 @@ IF ( USE_SYSTEM_LIBRARY_PATHS )
-IF ( ${OPENVDB_ABI_VERSION_NUMBER} )
-  MESSAGE( STATUS "Using openvdb abi version ${OPENVDB_ABI_VERSION_NUMBER}")
+# Configure DCC installation if necessary
 
-  ADD_DEFINITIONS (-DOPENVDB_ABI_VERSION_NUMBER=${OPENVDB_ABI_VERSION_NUMBER} )
-ELSEIF ( OPENVDB_ENABLE_3_ABI_COMPATIBLE )
+IF ( OPENVDB_BUILD_HOUDINI_PLUGIN )
+  SET ( USE_HOUDINI ON )
+ENDIF ()
 
-  MESSAGE( DEPRECATION "OPENVDB_ENABLE_3_ABI_COMPATIBLE is deprecated. Instead use OPENVDB_ABI_VERSION_NUMBER=N, where N is the abi version.")
+IF ( OPENVDB_BUILD_MAYA_PLUGIN )
+  SET ( USE_MAYA ON )
+ENDIF ()
 
-  ADD_DEFINITIONS ( -DOPENVDB_3_ABI_COMPATIBLE )
+IF ( USE_MAYA AND USE_HOUDINI )
+  # @todo technically this is possible so long as library versions match
+  # exactly but it's difficult to validate and dangerous
+  MESSAGE ( FATAL_ERROR "Cannot build both Houdini and Maya plugins against "
+    "the same core dependencies. Plugins must be compiled separately to "
+    "ensure the required DCC dependencies are met."
+    )
 ENDIF ()
 
-IF ( OPENVDB_CXX_STRICT )
-    # Add definitions for a number of compiler warnings for GCC and Clang for all sub projects
-    IF ( "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" )
-        MESSAGE ( STATUS "Configuring Clang CXX warnings" )
-        ADD_DEFINITIONS (
-            -Wall
-            -Wextra
-            -Wconversion
-            -Wno-sign-conversion
-        )
-    ELSEIF ( "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" )
-        MESSAGE ( STATUS "Configuring GCC CXX warnings" )
-        ADD_DEFINITIONS (
-            -Wall
-            -Wextra
-            -pedantic
-            -Wcast-align
-            -Wcast-qual
-            -Wconversion
-            -Wdisabled-optimization
-            -Woverloaded-virtual
-        )
-    ELSE ()
-        MESSAGE ( WARNING "No available CXX warnings for compiler "${CMAKE_CXX_COMPILER_ID} )
+# Configure component dependencies by loading the Houdini/Maya setup
+# scripts. These also find the Houdini/Maya installations
+
+IF ( USE_HOUDINI )
+  INCLUDE ( OpenVDBHoudiniSetup )
+ENDIF ()
+
+IF ( USE_MAYA )
+  INCLUDE ( OpenVDBMayaSetup )
+ENDIF ()
+
+#########################################################################
+
+# Configure OpenVDB Library and ABI versions
+
+IF ( NOT OPENVDB_BUILD_CORE )
+  # Find VDB installation and determine lib/abi versions
+  FIND_PACKAGE ( OpenVDB REQUIRED )
+  # Check ABI version was found and explicitly error if attempting to build against
+  # an incompatible Houdini version
+  IF ( OpenVDB_ABI AND OPENVDB_HOUDINI_ABI )
+    IF ( NOT ${OpenVDB_ABI} EQUAL ${OPENVDB_HOUDINI_ABI} )
+      MESSAGE ( FATAL_ERROR "Located OpenVDB installation is not ABI compatible with "
+        "Houdini Version ${Houdini_VERSION}. Requires ABI ${OPENVDB_HOUDINI_ABI}, found "
+        "ABI ${OpenVDB_ABI}." )
     ENDIF ()
+  ENDIF ()
+ELSE ()
+  # Only use find package to determine the VDB version in this directory
```

Comment:
The issue I'm having is related to these lines. Even if you're building the core library yourself, it still attempts to find the VDB version from an installed version of the library - I don't really understand why we'd want to do that?

As part of FindOpenVDB.cmake, there are a bunch of tests for dependent libraries which kick up these kind of warnings:

> warning: cannot resolve item 'libboost_iostreams.dylib'
> 
>   possible problems:
>     need more directories?
>     need to use InstallRequiredSystemLibraries?
>     run in install tree instead of build tree?
> 
> -- warning: gp_resolved_file_type non-absolute file 'libboost_iostreams.dylib' returning type 'other' -- possibly incorrect

I note that the build still appears to complete correctly, it's just that the cmake stage outputs a ton of confusing warnings and output when you have the core library already installed.

To reproduce:
* mkdir build
* cd build
* cmake -DOPENVDB_BUILD_HOUDINI_PLUGIN=ON -DUSE_BLOSC=ON -DOPENVDB_CORE_STATIC=OFF ../.
* make -j
* make install
* rm -rf *
* cmake -DOPENVDB_BUILD_HOUDINI_PLUGIN=ON -DUSE_BLOSC=ON -DOPENVDB_CORE_STATIC=OFF ../.